### ISSUE TITLE
Adding contact gives "Failed to find the following users"

### ISSUE THREAD

**Issue by singpolyma** [vector-im/element-web#21499]
### Steps to reproduce

1. Click "+" to add a direct message
2. Paste `@_bifrost_=2b14165551234=40cheogram.com:aria-net.org` in the box

#### What did you expect?

Should allow me to add the contact.

#### What happened instead?

Got error:

```
Failed to find the following users

The following users might not exist or are invalid, and cannot be invited: @_bifrost_=2b14165551234=40cheogram.com:aria-net.org
```

Typing `@` and then pasting `_bifrost_=2b14165551234=40cheogram.com:aria-net.org` works as expected. Only pasting the whole string (or using a matrix.to link) fails.

### Operating system

Debian

### Browser information

Firefox 91.6.0esr

### URL for webapp

https://app.element.io

### Application version

Element version: 1.10.7 Olm version: 3.2.8

### Homeserver

matrix.org

### Will you send logs?

Yes
